[ONNX] minor doc improvements and cleanup (#62514)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #64382 [ONNX] Deprecate use_external_data_format param from torch.onnx.export() function. (#62257)
* #64380 [ONNX] Deprecated the example_outputs param from torch.onnx.export() function. (#62815)
* **#64373 [ONNX] minor doc improvements and cleanup (#62514)**
* #64372 [ONNX] Add supplementary tests and description for custom_opsets param from torch.onnx.export() function. (#62085)
* #64371 [ONNX] Remove strip_doc_string param from torch.onnx.export() function. (#61712)
* #64370 [ONNX] Remove argument _retain_param_name from torch.onnx.export() function. (#61702)

* Fix some bad formatting and clarify things in onnx.rst.
* In `export_to_pretty_string`:
    * Add documentation for previously undocumented args.
    * Document that `f` arg is ignored and mark it deprecated.
    * Update tests to stop setting `f`.
    * Warn if `_retain_param_name` is set.
* Use double quotes for string literals in test_operators.py.

Differential Revision: [D30905271](https://our.internmc.facebook.com/intern/diff/D30905271)

garymm committed:
[ONNX] minor doc improvements and cleanup (#62514)

* Fix some bad formatting and clarify things in onnx.rst.
* In `export_to_pretty_string`:
    * Add documentation for previously undocumented args.
    * Document that `f` arg is ignored and mark it deprecated.
    * Update tests to stop setting `f`.
    * Warn if `_retain_param_name` is set.
* Use double quotes for string literals in test_operators.py.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] minor doc improvements and cleanup (#62514)"


* Fix some bad formatting and clarify things in onnx.rst.
* In `export_to_pretty_string`:
    * Add documentation for previously undocumented args.
    * Document that `f` arg is ignored and mark it deprecated.
    * Update tests to stop setting `f`.
    * Warn if `_retain_param_name` is set.
* Use double quotes for string literals in test_operators.py.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] minor doc improvements and cleanup (#62514)"


* Fix some bad formatting and clarify things in onnx.rst.
* In `export_to_pretty_string`:
    * Add documentation for previously undocumented args.
    * Document that `f` arg is ignored and mark it deprecated.
    * Update tests to stop setting `f`.
    * Warn if `_retain_param_name` is set.
* Use double quotes for string literals in test_operators.py.

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] minor doc improvements and cleanup (#62514)"


* Fix some bad formatting and clarify things in onnx.rst.
* In `export_to_pretty_string`:
    * Add documentation for previously undocumented args.
    * Document that `f` arg is ignored and mark it deprecated.
    * Update tests to stop setting `f`.
    * Warn if `_retain_param_name` is set.
* Use double quotes for string literals in test_operators.py.

[ghstack-poisoned]

msaroufim commented:
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30905271).

BowenBao committed:
Update on "[ONNX] minor doc improvements and cleanup (#62514)"


* Fix some bad formatting and clarify things in onnx.rst.
* In `export_to_pretty_string`:
    * Add documentation for previously undocumented args.
    * Document that `f` arg is ignored and mark it deprecated.
    * Update tests to stop setting `f`.
    * Warn if `_retain_param_name` is set.
* Use double quotes for string literals in test_operators.py.

Differential Revision: [D30905271](https://our.internmc.facebook.com/intern/diff/D30905271)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30905271).

BowenBao committed:
Update on "[ONNX] minor doc improvements and cleanup (#62514)"


* Fix some bad formatting and clarify things in onnx.rst.
* In `export_to_pretty_string`:
    * Add documentation for previously undocumented args.
    * Document that `f` arg is ignored and mark it deprecated.
    * Update tests to stop setting `f`.
    * Warn if `_retain_param_name` is set.
* Use double quotes for string literals in test_operators.py.

Differential Revision: [D30905271](https://our.internmc.facebook.com/intern/diff/D30905271)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30905271).
